docs/docs/working-with-video.md

```diff
@@ -0,0 +1,156 @@
+```
+
+## Querying video data from Markdown with GraphQL
+
+If a Markdown page or post has a featured video, you might want to include a video URL and title in [its frontmatter](/docs/adding-markdown-pages#note-on-creating-markdown-files). That makes it easy to pass those values into our custom component:
+
+```markdown:title=my-first-post.md
+---
+path: "/blog/my-first-post"
+date: "2019-03-27"
+title: "My first blog post"
+videoSourceURL: https://www.youtube.com/embed/dQw4w9WgXcQ
+videoTitle: "Official Music Video on YouTube"
+---
+```
+
+To include a video component in a template, you could start with something like this:
+
+```jsx:title=vlog-template.js
+import React from "react"
+import { graphql } from "gatsby"
+
+import Video from "../components/video"
+
+export default function VlogTemplate({
+  data, // this prop will be injected by the GraphQL query below.
+}) {
+  const { markdownRemark } = data // data.markdownRemark holds our post data
+  const { frontmatter, html } = markdownRemark
+  return (
+    <div className="blog-post-container">
+      <div className="blog-post">
+        <h1>{frontmatter.title}</h1>
+        <h2>{frontmatter.date}</h2>
+        <Video
+          videoSrcURL={frontmatter.videoSrcURL}
+          videoTitle={frontmatter.videoTitle}
+        />
+        <div
+          className="blog-post-content"
+          dangerouslySetInnerHTML={{ __html: html }}
+        />
+      </div>
+    </div>
+  )
+}
+
+export const pageQuery = graphql`
+  query($path: String!) {
+    markdownRemark(frontmatter: { path: { eq: $path } }) {
+      html
+      frontmatter {
+        date(formatString: "MMMM DD, YYYY")
+        path
+        title
+        videoSrcURL
+        videoTitle
+      }
+    }
+  }
+`
+```
+
+## Hosting your own HTML5 video files
+
+It's super common to source video from YouTube, Twitch or Vimeo. But what if you want to host your own video and include it as HTML5 video?
+
+> This discussion is ongoing on Github, chime in with your ideas here: https://github.com/gatsbyjs/gatsby/issues/3346
+
+To include your own video files that will work in multiple web browsers and platforms, you'll need to read up a bit on video extensions and codecs. We recommend MDN as a resource: <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Supported_media_formats">Media formats for HTML audio and video</a>. You may need video converter software to produce the necessary formats to support a range of devices and environments, such as `.webm` and `.mp4`.
+
+## Using custom video players
+
+One advantage of integrating a custom component with your own hosted video is it can give you more control over the video player, including its accessibility. It is strongly encouraged to provide captions and subtitles for your videos, and use a player with accessible controls.
+
+Check out the accessible HTML5 video player for React from Paypal: https://github.com/paypal/accessible-html5-video-player#react-version
```

Comment:
Should this be linkified?

Reply:
I thought Markdown would convert that automatically, but I will check! Thanks for the tip.

Reply:
It will — I should have clarified. I meant turning it into a named link, e.g. `"PayPal has created an [accessible HTML5 video player](#link)."`